<launch>
  <param name ="/use_sim_time" value="true"/>

  <arg name="model" default="/home/beoms/catkin_ws/src/ese_urdf/urdf/esevelo.urdf"/>
  <arg name="gui" default="true" />
  <arg name="rvizconfig" default="$(find urdf_tutorial)/rviz/urdf.rviz" />

  <param name="robot_description" textfile="$(arg model)" />
  <!--param name="use_gui" value="$(arg gui)"/ -->

  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" /> 
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />

  <node pkg = "rviz" name = "rviz" type = "rviz"
          required = "true" args = "-d $(find ese_lidar_projection)/rviz/localization.rviz"/>

  <node name="ese_lidar_projection" pkg="ese_lidar_projection" type="ese_lidar_projection" output="screen">
    <param name="ang_threshold" value="2.0"/>

    <param name="ground_threshold" value="1.0"/>
    <param name="objt_threshold" value="0.7"/>

    <param name="h_intensity_threshold" value="1000.0"/>
    <param name="l_intensity_threshold" value="12.0"/>

    <param name="lane_height_threshold" value="3.3"/>
  </node>

  <node pkg="ese_slam" name="ese_localization" type="ese_localization" output="screen">
    <param name="mapSize" value="15.0"/>
    <param name="targetMapSize" value="65.0"/>
    <param name="search_size" value="60.0"/>

    <param name="objt_leafSize" value="2.0"/>
    <param name="lane_leafSize" value="0.05"/>
  </node>

  <node pkg = "rosbag" name = "rosbag" type = "play"
          args = "--clock --pause /home/beoms/Desktop/bag_file/velo_0315/filter.bag"/> 

</launch>